configure

```diff
@@ -29,9 +29,14 @@ ERLANG_MD5="false"
 SKIP_DEPS=0
 
 COUCHDB_USER="$(whoami 2>/dev/null || echo couchdb)"
-SM_VSN=${SM_VSN:-"1.8.5"}
 ARCH="$(uname -m)"
 
+# Try to detect spidermonkey (libmozjs) version automatically
+SM_VSN=${SM_VSN:-$(/sbin/ldconfig -p | sed -n 's/.libmozjs-*\([0-9]*\).*/\1/p' | head -n 1)}
+if [ "$SM_VSN" = "185" ]; then
+  SM_VSN="1.8.5"
+fi
+
```

Comment:
We need this to fail gracefully if it can't find SpiderMonkey. Currently I'm finding that `./configure` will fail with a line like

> Unsupported SpiderMonkey version: 

which comes from `src/couch/rebar.config.script`. I don't have a specific suggestion of where to put the improvement but I think it would be good for the system to tell the user that it could not find a SpiderMonkey library  and bail a little more clearly

Reply:
If the version cannot be detected, the variable `SM_VSN` will be empty. I added an `elif` case exiting the configure script and warning the user. I hope that's good enough.

Reply:
Yep, looks good except for the typos ("SpiderMonkey", "explicitly")